syft/src/main/java/org/openmined/syft/execution/SyftJob.kt

```diff
@@ -125,25 +132,48 @@ class SyftJob internal constructor(
         plans: ConcurrentHashMap<String, Plan>,
         clientConfig: ClientConfig,
         dataLoader: DataLoader,
-        trainingParameters: TrainingParameters
+        trainingParameters: TrainingParameters,
+        checkPointSerializer: CheckPointSerializer? = null
```

Comment:
Can we pass this in the constructor?
Also, why `clientConfig` had to be passed here and not in the constructor? Can a job execute different plans with different client configurations?

Reply:
because `clientConfig` will only be available when the job is on ReadyStatus